tests/test_filter_count.py

```diff
@@ -181,43 +271,115 @@ def test_prune_on_small_data(filter_by):
-    assert len(data["labels"]) == int(round(np.sum(calibrated_cj)))
+    assert len(dataset["labels"]) == int(round(np.sum(calibrated_cj)))
 
     calibrated_cj2 = count.compute_confident_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
         calibrate=True,
     )
 
     # Check equivalency
     assert np.all(calibrated_cj == calibrated_cj2)
 
 
+def test_calibrate_joint_multilabel():
+    dataset = multilabel_data
+    cj = count.compute_confident_joint(
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        multi_label=True,
+        calibrate=False,
+    )
+    calibrated_cj = count.calibrate_confident_joint(
+        confident_joint=cj,
+        labels=dataset["labels"],
+        multi_label=True,
+    )
+    y_one = int2onehot(dataset["labels"])
+    # Check calibration
+    for class_num in range(0, len(calibrated_cj)):
+        label_counts = np.bincount(y_one[:, class_num])
+        assert all(calibrated_cj[class_num].sum(axis=1).round().astype(int) == label_counts)
+        assert len(dataset["labels"]) == int(round(np.sum(calibrated_cj[class_num])))
+    calibrated_cj2 = count.compute_confident_joint(
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        calibrate=True,
+        multi_label=True,
+    )
+    # Check equivalency
+    assert np.all(calibrated_cj == calibrated_cj2)
+
+
 @pytest.mark.parametrize("use_confident_joint", [True, False])
 def test_estimate_joint(use_confident_joint):
+    dataset = data
     joint = count.estimate_joint(
-        labels=data["labels"],
-        pred_probs=data["pred_probs"],
-        confident_joint=data["cj"] if use_confident_joint else None,
+        labels=dataset["labels"],
+        pred_probs=dataset["pred_probs"],
+        confident_joint=dataset["cj"] if use_confident_joint else None,
     )
 
     # Check that joint sums to 1.
     assert abs(np.sum(joint) - 1.0) < 1e-6
 
 
+@pytest.mark.parametrize("use_confident_joint", [True, False])
+def test_estimate_joint_multilabel(use_confident_joint):
+    dataset = multilabel_data
+    if use_confident_joint:
+        cj = count.compute_confident_joint(
+            labels=dataset["labels"], pred_probs=dataset["pred_probs"], multi_label=True
+        )
+        joint = count.estimate_joint(
+            labels=dataset["labels"],
+            pred_probs=dataset["pred_probs"],
+            confident_joint=cj,
+            multi_label=True,
+        )
+
+    else:
+
+        joint = count.estimate_joint(
+            labels=dataset["labels"],
+            pred_probs=dataset["pred_probs"],
+            confident_joint=dataset["cj"] if use_confident_joint else None,
+            multi_label=True,
+        )
+    # Check that each joint sums to 1.
+    for j in joint:
+        assert abs(np.sum(j) - 1.0) < 1e-6
+
```

Comment:
We should also be checking that output of `estimate_joint` is the same whether or not `confident_joint` was passed in as precomputed object or computed internally in this function. The current test does not appear to check the values of the output beyond that they sum to one.  

I am not sure you can implement my suggestion via `parametrize()`, may have to explicitly call `estimate_joint` multiple times to compare the outputs with each other.

Reply:
OK

Reply:
I've added a separate test case, the one that call it multiple times

Reply:
https://github.com/aditya1503/cleanlab/blob/84b4f960e23c1562dbf254aa8c37dc6544f93d80/tests/test_filter_count.py#L344